rlog/2024-06-24-gsub-largemsg-improvements.mdx

```diff
@@ -0,0 +1,396 @@
+It also raises the probability of simultaneous redundant transmissions to the same peer, 
+resulting in bandwidth wastage, congestion, and slow message propagation to the network. 
+Moreover, the benefits of sequential message relaying can be compromised by prioritizing slow (or busy) peer(s).
+
+7. Most use cases necessitate validating received messages before forwarding them to the next-hop peers.
+For a higher message transfer time $(\tau )$, this store-and-forward delay accumulates across the hops traveled by the message.  
+
+## Possible Improvements
+
+### 1. Minimizing Transfer Time for Large Messages
+The impact of message size and achievable data rate on message transmit time $\tau$ is crucial, 
+as this time accumulates due to the store-and-forward delay introduced at intermediate hops. 
+Some possible improvements minimizing overall message dissemination latency include:
+
+#### a. Message Fragmentation
+In a homogeneous network, network-wide message dissemination time (ignoring any processing delays) 
+can be simplified to roughly $\delta \approx \delta_{Tx} + P_h$, where $\delta_{Tx}$ represents accumulative message transmit time denoted as $\delta_{Tx} = \frac{S}{R} \times h$, 
+with $S$ being the data size, and $h$ being the number of hops in the longest path. 
+
+Partitioning a large message into n fragments reduces a single fragment transmit time to $\frac{\delta_{Tx}}{n}$. 
+As a received fragment can be immediately relayed by the receiver (while the sender is still transmitting the remaining fragments), 
+it reduces the transmit delay to $\delta_{Tx} = \frac{S}{R} \times \frac{2h-1}{n}$. 
+
+This time reduction is mainly attributed to the smaller store-and-forward delay involved in fragment transmissions.
+
+However, it is worth noting that many applications require each fragment to be individually verifiable.
+At the same time, message fragmentation allows a malicious peer to never relay some fragments of a message, 
+which can lead to a significant rise in the application's receive buffer size. 
+
+Therefore, message fragmentation requires a careful tradeoff analysis between time and risks.
+
+#### b. Message Staggering
+Considering the same bandwidth, the time $\tau_D$ required for sending a message to D peers stays the same, 
+even if we relay to all peers in parallel or send sequentially to the peers, i.e., $\tau_D = \sum_{i=1}^{D} \tau_i$. 
+
+However, sequential relaying results in quicker message reception at individual peers ($\tau_1 \ll \tau_D$) due to bandwidth saturation for a particular peer. 
+So, the receiver can start relaying early to its mesh members while the original sender is still sending it to other peers. 
+
+As a result, after every $\frac{\tau_D}{D}$ milliseconds, 
+the number of peers receiving the message increases by $2^r\ \forall\ r \lt D$ and by $\sum_{r=0}^{D-1} \lambda_{r-1}\ \forall\ r \geq D$. 

```

Comment:
Could you clarify why 2^r is used when r < D?

Reply:
During the first D rounds, every round doubles the peers that have received the message (triggering double transmissions in the following round). After that, all mesh members for the peers involved in the first round are covered.